nifi-nar-bundles/nifi-smb-bundle/nifi-smb-processors/src/main/java/org/apache/nifi/processors/smb/ListSmb.java

```diff
@@ -0,0 +1,353 @@
+                        + "coming from different file shares considered to be identical."),
+        @WritesAttribute(attribute = "timestamp", description =
+                "The timestamp of when the file's content in the filesystem as 'yyyy-MM-dd'T'HH:mm:ssZ'"),
+        @WritesAttribute(attribute = "createTime", description =
+                "The timestamp of when the file was created in the filesystem as 'yyyy-MM-dd'T'HH:mm:ssZ'"),
+        @WritesAttribute(attribute = "lastAccessTime", description =
+                "The timestamp of when the file was accessed in the filesystem as 'yyyy-MM-dd'T'HH:mm:ssZ'"),
+        @WritesAttribute(attribute = "changeTime", description =
+                "The timestamp of when the file's attributes was changed in the filesystem as 'yyyy-MM-dd'T'HH:mm:ssZ'"),
+        @WritesAttribute(attribute = "size", description = "The number of bytes in the source file"),
+        @WritesAttribute(attribute = "allocationSize", description = "The number of bytes allocated for the file on the server"),
+})
+@Stateful(scopes = {Scope.CLUSTER}, description =
+        "After performing a listing of files, the state of the previous listing can be stored in order to list files "
+                + "continuously without duplication."
+)
+public class ListSmb extends AbstractListProcessor<SmbListableEntity> {
+
+    public static final PropertyDescriptor DIRECTORY = new PropertyDescriptor.Builder()
+            .displayName("Input Directory")
+            .name("directory")
+            .description("The network folder to which files should be written. This is the remaining relative " +
+                    "after the hostname: smb://HOSTNAME:PORT/SHARE/[DIRECTORY]\\sub\\directories. It is also possible "
+                    + " to add subdirectories using this property. The given path on the remote file share must exists. "
+                    + "The existence of the remote folder can be checked using verification. You may mix different "
+                    + "directory separators in this property. If so NiFi will unify all of them and will use windows's"
+                    + "directory separator: '\\' ")
+            .required(false)
+            .addValidator(NON_BLANK_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor MINIMUM_AGE = new PropertyDescriptor.Builder()
+            .displayName("Minimum File Age in milliseconds")
+            .name("min-file-age")
+            .description(
+                    "Any file younger then the given value will be omitted. Ideally this value should be greater then"
+                            + "the amount of time needed to perform a list.")
+            .required(true)
+            .addValidator(TIME_PERIOD_VALIDATOR)
+            .defaultValue("5ms")
+            .build();
+
+    public static final PropertyDescriptor MINIMUM_SIZE = new PropertyDescriptor.Builder()
+            .displayName("Minimum File Size in bytes")
```

Comment:
Corresponding to change the size validation, this can be changed:
```suggestion
            .displayName("Minimum File Size")
```